app/recipe/serializer.py

```diff
@@ -0,0 +1,33 @@
+from rest_framework import serializers
+from recipe.models import Ingredient, Recipe
+
+class IngredientSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Ingredient
+        fields = ['name']
+        read_only_fields = ('id',)
+
+
+class RecipeSerializer(serializers.ModelSerializer):
+    ingredients = IngredientSerializer(many=True)
+    class Meta:
+        model = Recipe
+        fields = ('id', 'name', 'description', 'ingredients')
+        read_only_fields = ('id',)
+
+    def create(self, validated_data):
+        ingredient_data = validated_data.pop('ingredients')
+        recipe = Recipe.objects.create(**validated_data)
+        for ingredient in ingredient_data:
+            Ingredient.objects.create(recipe=recipe, **ingredient)
+        return recipe
+
+    def update(self, instance, validated_data):
+        ingredient_data = validated_data.pop('ingredients')
+        for (key, value) in validated_data.items():
+            setattr(instance, key, value)
```

Comment:
Interesting approach. FYI to update the instance we can leverage the default update implementation by calling the same method in the parent like this:
```super().create(validated_data)```

This could replace lines 27,28 and 32